docs/content/integrations/dagstermill/using-notebooks-with-dagster.mdx

```diff
@@ -0,0 +1,394 @@
+
+### Step 5.3: Modify the notebook
+
+We need to make a small change in our Jupyter notebook to allow Dagster to supply the `iris_dataset` asset as input. Behind the scenes, Dagster uses `papermill` to inject parameters into notebooks. `papermill` works by replacing a notebook cell with the `parameters` tag with a custom cell that can fetch the desired data.
+
+To accomplish this, we need to tag the cell in the `iris-kmeans` notebook that fetches the Iris dataset. This allows us to replace the cell with the data-fetching logic that loads the `iris_dataset` asset and retain the ability to run the Jupyter notebook in a standalone context. We'll cover this in more detail later in the tutorial.
+
+To add the `parameters` tag, you may need to turn on the display of cell tags in Jupyter:
+
+1. In Jupyter, navigate to **View > Cell Toolbar > Tags**:
+
+   <Image
+   src="/images/integrations/dagstermill/jupyter-view-menu.png"
+   width={1056}
+   height={640}
+   />
+
+2. Click **Add Tag** to add a `parameters` tag:
+
+   <Image
+   src="/images/integrations/dagstermill/jupyter-tags.png"
+   width={2278}
+   height={466}
+   />
+
+---
+
+## Step 6: Materialize the assets
+
+Next, we'll materialize our `iris_dataset` and notebook assets.
+
+1. In Dagit, open the **Asset Graph** page.
+
+2. Click the **Reload definitions** button. This ensures that Dagit picks up the changes you made in the previous steps.
+
+   At this point, the `iris_dataset` asset should display above the `iris_kmeans_jupyter` asset as an upstream dependency:
+
+   <Image
+   src="/images/integrations/dagstermill/dagit-three.png"
+   width={2870}
+   height={1398}
+   />
+
+3. Click the **Materialize all** button near the top right corner of the page, which will launch a run to materialize the assets.
+
+That's it! You now have working Jupyter and Dagster assets!
+
+---
+
+## Extra credit: Fetch a Dagster asset in a Jupyter notebook
+
+What if you want to do additional analysis of the Iris dataset and create a new notebook? How can you accomplish this without duplicating code or re-fetching data?
+
+The answer is simple: use the `iris_dataset` Dagster asset!
+
+In the Jupyter notebook, import the Dagster repository and use the <PyObject object="RepositoryDefinition" method="load_asset_value" /> function to load the data for the `iris_dataset` asset we created in [Step 5.1: Create the Iris dataset asset](#step-51-create-the-iris-dataset-asset):
+
+```python
+from tutorial_template import template_tutorial
+
+iris = template_tutorial.load_asset_value("iris_dataset")
+```
+
+Then, whenever you run the notebook using Jupyter, you'll be able to work with the `iris_dataset` asset:
+
+```shell
+jupyter notebook /path/to/new/notebook.ipynb
+```
+
+Behind the scenes, when `load_asset_value` is called, Dagster fetches the value of `iris_dataset` that was most recently materialized and stored by an I/O manager.
```

Comment:
@yuhan hows this? should there be more explanation?

Reply:
lg2m